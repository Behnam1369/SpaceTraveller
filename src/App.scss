body {
  header {
    padding: 30px;

    .logo {
      width: 40px;
      height: 40px;
    }

    nav {
      float: right;
    }

    nav > a {
      text-decoration: none;
      color: blue;
      margin: 20px;
    }

    nav > a.active {
      text-decoration: underline;
      color: red;
    }
  }

  .missionList {
    width: 90%;
    margin: auto;
    border-collapse: collapse;

    thead {
      th {
        border: solid 1px silver;
        font-family: Helvetica, sans-serif;
        padding: 5px;
      }
    }

    tbody tr:nth-child(odd) {
      background-color: whitesmoke;
    }

    td {
      border: solid 1px silver;
      font-family: Helvetica, sans-serif;
      padding: 5px;

      span {
        background-color: gray;
        color: white;
        white-space: nowrap;
        padding: 5px;
        font-family: Helvetica, sans-serif;
        font-size: 14px;
        border-radius: 3px;
      }

      span.joined {
        background-color: darkcyan;
      }
    }

    td:nth-child(1) {
      font-weight: bold;
      text-align: center;
    }

    td:nth-child(2) {
      font-family: Arial, sans-serif;
      text-align: left;

      button {
        background-color: transparent;
        border: none;
        color: blue;
        cursor: pointer;
      }
    }

    td:nth-child(4) button {
      background-color: white;
      color: gray;
      border: solid 1px gray;
      white-space: nowrap;
      padding: 5px;
      font-family: Helvetica, sans-serif;
      font-size: 14px;
      cursor: pointer;
      border-radius: 4px;
    }

    td:nth-child(4) button.joined {
      color: red;
      border-color: red;
    }
  }

  .profile {
    margin: 7%;

    h1 {
      font-family: Helvetica, sans-serif;
    }

    h2 {
      font-family: Helvetica, sans-serif;
    }

    ul {
      width: 50%;
      margin: 0;
      padding: 0;

      li {
        list-style: none;
        margin: 0;
        padding: 20px;
        border: solid 1px lightgray;
      }
    }
  }
}
