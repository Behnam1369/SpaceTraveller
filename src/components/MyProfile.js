import { React } from 'react';
import { useSelector } from 'react-redux';

function MyProfile() {
  const { missions } = useSelector((state) => state.missions);
  let joinedMissions = [];
  if (missions !== null) {
    joinedMissions = missions.filter((el) => el.joined);
  }
  const rockets = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
  return (
    <div className="profile">
      <h1>Profile</h1>
      <div className="divs-const">
        <div className="profile-categories">
          {joinedMissions.length > 0
            ? (
              <>
                <h2>My Missions</h2>
                <ul>
                  {joinedMissions.map((el) => (
                    <li key={el.mission_id}>
                      {' '}
                      {el.mission_name}
                      {' '}
                    </li>
                  ))}
                </ul>
              </>
            )
            : <h2>You have not joined any missions yet.</h2>}
        </div>
        <div className="profile-categories">
          {reservedRockets.length > 0
            ? (
              <>
                <h2>My Rocket Reservations</h2>
                <ul>
                  {reservedRockets.map((el) => (
                    <li key={el.id}>
                      {' '}
                      {el.name}
                      {' '}
                    </li>
                  ))}
                </ul>
              </>
            )
            : <h2>You have not reserved any rocket yet.</h2>}
        </div>
      </div>
    </div>
  );
}

export default MyProfile;
