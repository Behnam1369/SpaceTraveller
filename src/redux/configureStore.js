import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import missionReducer from './missions';
import { rocketsReducer } from './rockets';

const rootReducer = combineReducers({
  rockets: rocketsReducer,
  missions: missionReducer,
});

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
