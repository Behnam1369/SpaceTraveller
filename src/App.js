import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Header from './components/Header';
import MyProfile from './components/MyProfile';
import './App.scss';
import MissionList from './components/MissionList';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Header />
        <Routes>
          <Route path="/MyProfile" element={<MyProfile />} />
          <Route path="/Missions" element={<MissionList />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
