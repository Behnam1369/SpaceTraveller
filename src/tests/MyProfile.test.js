import TestRenderer from 'react-test-renderer';
import {
  render,
  screen,
  fireEvent,
  waitFor,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import MyProfile from '../components/MyProfile';
import '@testing-library/jest-dom';
import configureStore from '../redux/configureStore';
import App from '../App';

describe('Testing MyProfile', () => {
  it('Testing MyProfile component snapshot', () => {
    const testRenderer = TestRenderer.create(
      <Provider store={configureStore}>
        <MyProfile />
      </Provider>,
    ).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
});

describe('Testing MyProfile component in Application', () => {
  it('Test showing the correct list of subscribed items', () => {
    const missions = [
      {
        mission_name: 'FIRST MISSION',
        mission_id: 'key1',
        description: 'TEST DESCRIPTION',
        reserved: false,
      },
      {
        mission_name: 'SECOND MISSION',
        mission_id: 'key2',
        description: 'ANOTHER TEST DESCRIPTION ',
        reserved: false,
        joined: true,
      },
    ];

    configureStore.dispatch({ type: 'space-traveller/missions/GET MISSIONS', missions });
    configureStore.dispatch({ type: 'book-store/books/SET LOADING', loading: false });
    render(
      <Provider store={configureStore}>
        <App />
      </Provider>,
    );
    fireEvent.click(screen.getByText('My Profile'));
    waitFor(() => screen.getByText('SECOND MISSION'));
    expect(screen.getByText('SECOND MISSION')).toBeInTheDocument();
  });
});
