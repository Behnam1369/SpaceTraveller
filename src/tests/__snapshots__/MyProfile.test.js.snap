// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MyProfile Testing MyProfile component snapshot 1`] = `
<div
  className="profile"
>
  <h1>
    Profile
  </h1>
  <div
    className="divs-const"
  >
    <div
      className="profile-categories"
    >
      <h2>
        You have not joined any missions yet.
      </h2>
    </div>
    <div
      className="profile-categories"
    >
      <h2>
        You have not reserved any rocket yet.
      </h2>
    </div>
  </div>
</div>
`;
